providers/base/api/handlers/api_query_handler.go

```diff
@@ -73,24 +78,20 @@ func NewAPIQueryHandler[K providertypes.ResponseKey, V providertypes.ResponseVal
 		return nil, fmt.Errorf("no logger specified for api query handler")
 	}
 
-	if requestHandler == nil {
-		return nil, fmt.Errorf("no request handler specified for api query handler")
-	}
-
-	if apiHandler == nil {
-		return nil, fmt.Errorf("no api data handler specified for api query handler")
-	}
-
 	if metrics == nil {
 		return nil, fmt.Errorf("no metrics specified for api query handler")
 	}
 
+	priceFetcher, err := NewRestAPIPriceFetcher(requestHandler, apiHandler, metrics, cfg, logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create price fetcher: %w", err)
+	}
```

Comment:
just a note: this looks good to me but i anticipate that this may require some changes to the orchestrator implementation to account for the fact that the API query handler now utilizes a single interface with various implementations.

Reply:
requestHandler RequestHandler,
this assumes rest atm but we can probably just accept a single price fetcher interface here instead of `RequestHandler` and `DataHandler`.

Reply:
What is the requested change here?

Reply:
good note